packages/invoices/src/OutboundShipment/DetailView/modals/BatchesTable.tsx

```diff
@@ -16,62 +19,117 @@ import {
   const { register } = useFormContext();
   const t = useTranslation();
   const d = useFormatDate();
 
+  const onChangeValue: React.ChangeEventHandler<HTMLInputElement> = event => {
+    const value = Math.max(Number(event.target.value), 0);
+    const newValue = Math.min(value, batch.availableNumberOfPacks);
+
+    onChange(batch.id, newValue, batch.packSize);
+  };
+
   const stockLineInputProps = register(batch.id, {
-    min: { value: 1, message: t('error.greater-than-zero-required') },
-    pattern: { value: /^[0-9]+$/, message: t('error.number-required') },
+    min: { value: 0, message: t('error.invalid-value') },
+    max: {
+      value: batch.availableNumberOfPacks,
+      message: t('error.invalid-value'),
+    },
+    pattern: { value: /^[0-9]+$/, message: t('error.invalid-value') },
+    onChange: onChangeValue,
   });
 
-  const onChangeValue: React.ChangeEventHandler<HTMLInputElement> = event =>
-    onChange(batch.id, Number(event.target.value));
+  const expiryDate = new Date(batch.expiryDate);
+  const isDisabled = batch.availableNumberOfPacks === 0 || batch.onHold;
 
   // TODO format currency correctly
   return (
     <TableRow>
       <BasicCell align="right">{label}</BasicCell>
       <BasicCell sx={{ width: '88px' }}>
-        <NumericTextInput
-          {...stockLineInputProps}
-          sx={{ height: '32px' }}
-          disabled={batch.availableNumberOfPacks === 0}
-          onChange={onChangeValue}
+        <ModalNumericInput
+          inputProps={stockLineInputProps}
+          disabled={isDisabled}
         />
       </BasicCell>
-      <BasicCell>
-        <Checkbox disabled={batch.availableNumberOfPacks === 0} />
+      <BasicCell align="right">{batch.packSize}</BasicCell>
+      <BasicCell sx={{ width: '88px' }}>
+        <ReadOnlyInput number {...register(`${batch.id}_total`)} />
       </BasicCell>
-      <BasicCell align="right">{batch.totalNumberOfPacks}</BasicCell>
       <BasicCell align="right">{batch.availableNumberOfPacks}</BasicCell>
-      <BasicCell align="right">{batch.packSize}</BasicCell>
+      <BasicCell align="right">{batch.totalNumberOfPacks}</BasicCell>
       <BasicCell>{batch.batch}</BasicCell>
-      <BasicCell>{d(new Date(batch.expiryDate))}</BasicCell>
-      <BasicCell align="right">${batch.costPricePerPack}</BasicCell>
+      <BasicCell
+        sx={{ color: isAlmostExpired(expiryDate) ? 'error.main' : undefined }}
+      >
+        {d(expiryDate)}
+      </BasicCell>
+      <BasicCell>{batch.location}</BasicCell>
       <BasicCell align="right">${batch.sellPricePerPack}</BasicCell>
+      <BasicCell align="center">
+        <Checkbox disabled checked={batch.onHold} />
+      </BasicCell>
       <BasicCell>
-        <InfoIcon
-          fontSize="small"
-          sx={{ color: theme => theme.palette.lightGrey }}
-        />
+        <Popper
+          content={
+            <Grid spacing={2} container sx={{ padding: '20px' }}>
```

Comment:
this is all a bit too far probably - was just wanting to demo something to start the discussion about what to show here. if anything, maybe we don't want the info icon after all?

Thought that we could have the API extended slightly to return invoice names and quantities of 'other' invoices, in order to detail where the other stock is allocated 🤷

Reply:
All good, nice to have something concrete in place to get feedback